sympy/utilities/tests/test_wester.py

```diff
@@ -735,7 +735,7 @@ def test_J16():
 
 @XFAIL
 def test_J17():
-    assert deltaintegrate(f((x + 2)/5)*DiracDelta((x - 2)/3) - g(x)*diff(DiracDelta(x - 1), x), (x, 0, 3))
+    raise NotImplementedError("deltaintegrate(f((x + 2)/5)*DiracDelta((x - 2)/3) - g(x)*diff(DiracDelta(x - 1), x), (x, 0, 3))")
```

Comment:
How about

```suggestion
    assert deltaintegrate(f((x + 2)/5)*DiracDelta((x - 2)/3) - g(x)*diff(DiracDelta(x - 1), x), (x, 0, 3)) is not None
```

(or better, assert that it equals the correct answer). 

XFAIL tests should ideally xpass once they start working.

Reply:
What you're saying makes sense and I fully agree. I think that an xfail test should not only xpass when working but should actually be a reasonable non-xfail test of the correct behaviour.

Happy to change it but if you look at the surrounding code though you'll see why I did it like this.

Reply:
I actually don't understand this test. Why is it testing the internal `deltaintegrate` function? For whatever reason, it doesn't work, even if you split up the integral, but `integrate` itself works just fine:

```
>>> integrate(f((x + 2)/5)*DiracDelta((x - 2)/3) - g(x)*diff(DiracDelta(x - 1), x), (x, 0, 3))
3*f(4/5) + Subs(Derivative(g(x), x), x, 1)
```

The Wester tests should be testing the public APIs. 

So I actually think we should change this test to a passing one that tests the above.

Reply:
```suggestion
    assert integrate(f((x + 2)/5)*DiracDelta((x - 2)/3) - g(x)*diff(DiracDelta(x - 1), x), (x, 0, 3)) == 3*f(S(4)/5) + Subs(Derivative(g(x), x), x, 1)
```